src/app/books/books.service.ts

```diff
@@ -0,0 +1,28 @@
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs';
+import { map, delay } from 'rxjs/operators';
+import { Book } from './book/book.model';
+
+const API_URL =
+  'https://api.nytimes.com/svc/books/v3/lists/current/hardcover-fiction.json';
+const API_KEY = 'dVh2I4dlVdJYaXYaOFYtHhkGYISksZFo';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class BooksService {
+  constructor(private readonly http: HttpClient) {}
+
+  getBooks(): Observable<Book[]> {
+    return this.http
+      .get<{ results: { books: Book[] } }>(API_URL, {
```

Comment:
Create type for results (also no need to write _results : Type_, you can just write what is expected return type)